azure/Guidelines.md

```diff
@@ -18,6 +20,50 @@ Teams building ARM Resource Providers (RPs) MUST follow the additional guidance
 
 ARM RPs are Azure Fundamentals requirement for Azure Services and ARM RP review is another mandatory review. Some of the guidance overlaps with general API review, but passing one review will generally make the other one go very quickly.
 
+### Advice for new services
+
+Developing a new service requires the development of at least 1 (management plane) API and potentially one or more additional (data plane) APIs.  When reviewing v1 service APIs, we see common advice provided during the review.
+
+> A **management plane** API is implemented through the Azure Resource Manager (ARM) and is used by subscription administrators.  A **data plane** API is used by developers to implement applications.  Rarely, a subset of operations may be useful to both administrators and users, in which case it should appear in both APIs.
+
+* Think about naming from the context of a **developer experience**.  
+  * Start with the "things" your API manipulates, then think about the operations that a developer needs to do to these "things".
+  * Keep the verbs present-tense.  Avoid the use of past or future tense in most cases.
+  * Avoid the use of generic names like "Object", "Job", "Task", "Operation" (for example - the list is not exhaustive).
+  * What happens to the names when the focus of the service expands?  It may be worth starting with a less generic name to avoid a breaking change later on.
+* Think about the code that a customer will write both before and after the REST API call.  How will a developer use this API in the canonical use case?
```

Comment:
I'd consider adding a point about improving "discoverability" or "authoring" of the payload - i.e. structure it in a way that is self explanatory/intelli-sensible rather than require to consult documentation. Two examples:
1. Arrays of polymorphic objects - (usually) can be replaced with multiple arrays of objects of a single type (instead of ```{ fruits: [] }``` you can have  ```{ apples: [], oranges: [], bananas: [] }```. Frequently those would also be processed differently anyway (otherwise they wouldn't be different)
2. When having strings which can accept any value but your service predefines some well known values (e.g. perf counter names) model them as "extensible enums" - this way they will be present in the schema and generated SDKs